### ISSUE TITLE
Database Media Storage : clearing image cache does not clear cached image files from database

### ISSUE THREAD

**Issue by gwharton** [magento/magento2#23936]
### Preconditions (*)
1. magento 2.3-develop

### Steps to reproduce (*)
1. Deploy 2.3-develop

```
Stores -> Configuration
    Advanced -> System
        Storage Configuration for Media
            Media Storage = Database
            Select Media Database = default_setup
            Synchronize
            Save Config
Catalog -> Products
    Add Product
        Name = Test
        Price = 111
        Images and Videos
            Upload image "a.jpg"
        Save & Close
```
Verify cached images exist in filesystem
```
www-data@dev:~/dev1$ find pub/media/catalog/product/cache -name *.jpg
pub/media/catalog/product/cache/6e14adab429f71aeb7223303a247c563/a/_/a.jpg
pub/media/catalog/product/cache/f62d29ead3f33fe6484fbdb6d3785dc1/a/_/a.jpg
pub/media/catalog/product/cache/be852cb44d0b54faab62ffaae6be8535/a/_/a.jpg
pub/media/catalog/product/cache/cec7ecf00f20929af760713af4400fed/a/_/a.jpg
pub/media/catalog/product/cache/37bfafdfb226dffa175577a00672b1ba/a/_/a.jpg
pub/media/catalog/product/cache/8b2dc6afd39ee28ff58514eb10993c8f/a/_/a.jpg
pub/media/catalog/product/cache/d975e82a925000253a32e9c4541ea68c/a/_/a.jpg
pub/media/catalog/product/cache/09c5a17c8da4235f70a8c07e75bb5ef0/a/_/a.jpg
pub/media/catalog/product/cache/9fda6c6842c3f655bbb560bcbe7ed6d5/a/_/a.jpg
pub/media/catalog/product/cache/d5f386a768d7e7f87cc4fb8a6c1ef14f/a/_/a.jpg
pub/media/catalog/product/cache/52952a29dad7e0f0b516b44cc9fe0760/a/_/a.jpg
pub/media/catalog/product/cache/06dfa18d55e0337076e1f5ddd2129562/a/_/a.jpg
pub/media/catalog/product/cache/faa05d146573dd135d15c4c56517ffdb/a/_/a.jpg
pub/media/catalog/product/cache/3ced2c8980660288e4ea38fe4970b4e9/a/_/a.jpg
pub/media/catalog/product/cache/2bdb51d37a492679108b7c6c17f34e97/a/_/a.jpg
pub/media/catalog/product/cache/4ed1686d54d03efa60ce0fb8e68e54e9/a/_/a.jpg
pub/media/catalog/product/cache/ee5242f5f0a2a47237a7610f0274bc15/a/_/a.jpg
pub/media/catalog/product/cache/8f18f3a902701264b776ead80189598e/a/_/a.jpg
pub/media/catalog/product/cache/39fe0b11cb432d7c2d562e3c5b5d5f43/a/_/a.jpg
pub/media/catalog/product/cache/34c2321368257a27e78b60f1cf77ce60/a/_/a.jpg
www-data@dev:~/dev1$
```
Verify cached images exist in database
```
mysql> mysql> select filename,directory from media_storage_file_storage where directory LIKE 'catalog/product/cache%';
+----------+------------------------------------------------------------+
| filename | directory                                                  |
+----------+------------------------------------------------------------+
| a.jpg    | catalog/product/cache/9fda6c6842c3f655bbb560bcbe7ed6d5/a/_ |
| a.jpg    | catalog/product/cache/f62d29ead3f33fe6484fbdb6d3785dc1/a/_ |
| a.jpg    | catalog/product/cache/d5f386a768d7e7f87cc4fb8a6c1ef14f/a/_ |
| a.jpg    | catalog/product/cache/52952a29dad7e0f0b516b44cc9fe0760/a/_ |
| a.jpg    | catalog/product/cache/ee5242f5f0a2a47237a7610f0274bc15/a/_ |
| a.jpg    | catalog/product/cache/cec7ecf00f20929af760713af4400fed/a/_ |
| a.jpg    | catalog/product/cache/09c5a17c8da4235f70a8c07e75bb5ef0/a/_ |
| a.jpg    | catalog/product/cache/39fe0b11cb432d7c2d562e3c5b5d5f43/a/_ |
| a.jpg    | catalog/product/cache/3ced2c8980660288e4ea38fe4970b4e9/a/_ |
| a.jpg    | catalog/product/cache/8f18f3a902701264b776ead80189598e/a/_ |
| a.jpg    | catalog/product/cache/06dfa18d55e0337076e1f5ddd2129562/a/_ |
| a.jpg    | catalog/product/cache/34c2321368257a27e78b60f1cf77ce60/a/_ |
| a.jpg    | catalog/product/cache/37bfafdfb226dffa175577a00672b1ba/a/_ |
| a.jpg    | catalog/product/cache/be852cb44d0b54faab62ffaae6be8535/a/_ |
| a.jpg    | catalog/product/cache/4ed1686d54d03efa60ce0fb8e68e54e9/a/_ |
| a.jpg    | catalog/product/cache/2bdb51d37a492679108b7c6c17f34e97/a/_ |
| a.jpg    | catalog/product/cache/d975e82a925000253a32e9c4541ea68c/a/_ |
| a.jpg    | catalog/product/cache/6e14adab429f71aeb7223303a247c563/a/_ |
| a.jpg    | catalog/product/cache/faa05d146573dd135d15c4c56517ffdb/a/_ |
| a.jpg    | catalog/product/cache/8b2dc6afd39ee28ff58514eb10993c8f/a/_ |
+----------+------------------------------------------------------------+
20 rows in set (0.00 sec)

mysql>
```

```
System -> Cache Management
    Flush Catalog Images Cache
```
Verify cached images no longer exist in filesystem
```
www-data@dev:~/dev1$ find pub/media/catalog/product/cache -name *.jpg
find: ‘pub/media/catalog/product/cache’: No such file or directory
www-data@dev:~/dev1$
```
Verify cached images no longer exist in database
```
mysql> select filename,directory from media_storage_file_storage where directory LIKE 'catalog/product/cache%';
+----------+------------------------------------------------------------+
| filename | directory                                                  |
+----------+------------------------------------------------------------+
| a.jpg    | catalog/product/cache/9fda6c6842c3f655bbb560bcbe7ed6d5/a/_ |
| a.jpg    | catalog/product/cache/f62d29ead3f33fe6484fbdb6d3785dc1/a/_ |
| a.jpg    | catalog/product/cache/d5f386a768d7e7f87cc4fb8a6c1ef14f/a/_ |
| a.jpg    | catalog/product/cache/52952a29dad7e0f0b516b44cc9fe0760/a/_ |
| a.jpg    | catalog/product/cache/ee5242f5f0a2a47237a7610f0274bc15/a/_ |
| a.jpg    | catalog/product/cache/cec7ecf00f20929af760713af4400fed/a/_ |
| a.jpg    | catalog/product/cache/09c5a17c8da4235f70a8c07e75bb5ef0/a/_ |
| a.jpg    | catalog/product/cache/39fe0b11cb432d7c2d562e3c5b5d5f43/a/_ |
| a.jpg    | catalog/product/cache/3ced2c8980660288e4ea38fe4970b4e9/a/_ |
| a.jpg    | catalog/product/cache/8f18f3a902701264b776ead80189598e/a/_ |
| a.jpg    | catalog/product/cache/06dfa18d55e0337076e1f5ddd2129562/a/_ |
| a.jpg    | catalog/product/cache/34c2321368257a27e78b60f1cf77ce60/a/_ |
| a.jpg    | catalog/product/cache/37bfafdfb226dffa175577a00672b1ba/a/_ |
| a.jpg    | catalog/product/cache/be852cb44d0b54faab62ffaae6be8535/a/_ |
| a.jpg    | catalog/product/cache/4ed1686d54d03efa60ce0fb8e68e54e9/a/_ |
| a.jpg    | catalog/product/cache/2bdb51d37a492679108b7c6c17f34e97/a/_ |
| a.jpg    | catalog/product/cache/d975e82a925000253a32e9c4541ea68c/a/_ |
| a.jpg    | catalog/product/cache/6e14adab429f71aeb7223303a247c563/a/_ |
| a.jpg    | catalog/product/cache/faa05d146573dd135d15c4c56517ffdb/a/_ |
| a.jpg    | catalog/product/cache/8b2dc6afd39ee28ff58514eb10993c8f/a/_ |
+----------+------------------------------------------------------------+
20 rows in set (0.00 sec)

mysql>
```
### Expected result (*)
1. When the image cache is cleared, all cached images are removed from filesystem and database

### Actual result (*)
1. Cached images are removed from filesystem only. They remain in the database.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-18735` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @gwharton. Thank you for your report. 
The issue has been fixed in magento/magento2#24088 by @gwharton in 2.3-develop branch
Related commit(s):
 - [2e3c5696266228ec2ebf5fd192bbb33febb4cff8](https://github.com/magento/magento2/commit/2e3c5696266228ec2ebf5fd192bbb33febb4cff8)

The fix will be available with the upcoming 2.3.4 release.
